README.md

```diff
@@ -636,3 +636,37 @@ Compared to gomod and npm, Cachito support for pip has restrictions and limitati
 not expect. For more details, see the [Cachito pip documentation](docs/pip.md).
 
 [nexus-docs]: https://help.sonatype.com/repomanager3
+
+### git-submodule
+
+Cachito will produce a bundle that is downloadable at `/api/v1/requests/<id>/download`. This
+bundle will contain the application source code in the `app` directory. When `git-submodule`
+is passed as a `pkg_managers` argument for any Cachito request, the available git submodules
+within the requested repo will also become available as part of the downloadable bundle. If the
+repo contains multiple submodules, we will fetch them all. Although, recursion is not supported
+and hence only one level of submodules will be fetched.
+
+If a path to a specific git submodule is provided as part of the `packages` configuration,
+we would fetch the submodule and then process it as a regular package.
+
+```bash
+{
+  "repo": "https://github.com/namespace/repo",
+  "ref": "abcdef...",
+  "pkg_managers": ["git-submodule", "pip"],
+  "packages": {
+    "pip": [
+      {"path": "cool_python_submodule"}
+    ]
+  }
+}
+```
```

Comment:
Would it be possible to provide a real example instead? Maybe you could link one of the repos you used for testing or create a new one (keep in mind that subpaths for gomod packages are currently not supported).

Reply:
What would be benefit from providing a real repo example? IMO, if I were user, I'd rather like the current example which clearly mentions the components of url too. e.g. namespace, repo

I could maybe update the request to look like, `"repo": "https://github.com/<repo-namespace>/<repo>.git"`

wdyt?

Reply:
What I had in mind was to use a real value for "repo" and "ref", so a user could actually go to the repo and see what's in it, or maybe a developer could submit the request locally and see how it works.

But IDK how valuable it would be, I personally don't need it, I'm trying to put myself in the shoes of someone who does not already know how it works :D